spi/data-plane-selector/data-plane-selector-spi/src/main/java/org/eclipse/dataspaceconnector/dataplane/selector/instance/DataPlaneInstance.java

```diff
@@ -24,6 +25,9 @@
  * Representations of a data plane instance. Every DPF has an ID and a URL as well as a number, how often it was selected,
  * and a timestamp of its last selection time. In addition, there are extensible properties to hold specific properties.
  */
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = DataPlaneInstanceImpl.class, name = "dataspaceconnector:dataplaneinstance")
+})
```

Comment:
Is this needed?

Reply:
Yes this is need since DataPlaneInstance is actually an Interface so Jackson have difficulty to deserialize a DataPlaneInstanceDocument when a subtype isn't specify.

Reply:
@Alexwilfriedo could you create an issue to switch from this interface->implementation design to a simpler "data class" one?

Reply:
Yes sure

Reply:
@Alexwilfriedo  if i remember correctly if the DataPlaneInstanceImpl is registered in the serialization context of Jackson there is no need of the subtype annotation since there is `@JsonTypeName` on  `DataPlaneInstanceImpl`